napari/_vispy/layers/labels.py

```diff
@@ -68,8 +74,14 @@
     // we get a different value:
     // - if it's the empty key, exit;
     // - otherwise, it's a hash collision: continue searching
-    while ((abs(found - t) > 1e-8) && (abs(found - empty) > 1e-8)) {
-        t = t + 1;
+    float initial_t = t;
+    int count = 0;
+    while ((abs(found - initial_t) > 1e-8) && (abs(found - empty) > 1e-8)) {
+        count = count + 1;
+        t = initial_t + float(count);
+        if (count >= color_count) {
+            return vec4(0);
+        }
```

Comment:
If the count is above 0 and we hit an empty field then we could assume that color is not defined. Did I'm correct?

Reply:
Yes. It is, and it is handled by while loop condition